module/zfs/dsl_scan.c

```diff
@@ -3025,8 +3025,10 @@ dsl_scan_async_block_should_pause(dsl_scan_t *scn)
 	if (zfs_recover)
 		return (B_FALSE);
 
-	if (scn->scn_visited_this_txg >= zfs_async_block_max_blocks)
+	if (zfs_async_block_max_blocks != 0 && 
```

Comment:
This code change does not appear to be relevant to this PR?

Reply:
Yes, this changes are of my previous PR which is accepted but not merged yet! I remove this for this now